flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -212,6 +213,9 @@
      */
     private volatile boolean isRunning;
 
+    /** Flag to mark that the state of current task was restored successfully. */
+    private volatile boolean restored;
```

Comment:
remove `volatile`? Either it doesn't need to be thread safe or we are doing something wrong in some test. Note it doesn't look like it's thread safe right now even with the `volatile`.